### PR TITLE
Issue32 Add pagination

### PR DESCRIPTION
**Pull request by buda1994**

### CODE REVIEW

**Comment by ArCiGoUnoSquare** [comment]
I removed it, @geoperez . I was using it in my old approach to display the footer

File: src/Grid/Pager.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  spacer: {
+    flex: 0
+  },
+  select: {
+    border: '1px solid #c3c3c3',
+    borderRadius: '5px'
+  }
+});
+
+const Pager = ({ handlePager, rowsPerPage, page, filteredRecordCount, classes, totalRecordCount }) => {
```
Comment (kadosh) [inline src/Grid/Pager.js]: Can wel call it Paginator?

File: src/Grid/Grid.js [hunk 2]
```diff
@@ -21,33 +23,56 @@ class Grid extends React.Component {
...
         this.setState({
           data: tbResponse.Payload,
+          totalRecordCount: tbResponse.TotalRecordCount || 0,
+          filteredRecordCount: tbResponse.FilteredRecordCount || 0,
           aggregate: tbResponse.Aggregate
         });
       });
+
+    this.search.debounce(600).subscribe(() => {
+      const { dataSource, rowsPerPage, page, searchText } = this.state;
```
Comment (geoperez) [inline src/Grid/Grid.js]: Try to reuse `refreshGrid` method

File: src/Grid/Grid.js [hunk 3]
```diff
@@ -70,20 +95,43 @@ class Grid extends React.Component {
       </TableBody>
     );
 
+    const paginator = (
+      <TableRow>
+        <Paginator
+          dataSource={dataSource}
+          rowsPerPage={rowsPerPage}
+          page={page}
+          filteredRecordCount={this.state.filteredRecordCount}
```
Comment (geoperez) [inline src/Grid/Grid.js]: Why don't spread them at line 75?

File: src/Grid/Paginator.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+const Paginator = ({ handlePager, rowsPerPage, page, filteredRecordCount, classes, totalRecordCount }) => {
+  const handleChangePage = (event, page) => {
+    handlePager(rowsPerPage, page);
+  };
+    
+  const handleChangeRowsPerPage = event => {
+    handlePager(event.target.value, page);
+  };
+
+  const message = totalRecordCount === filteredRecordCount ?
```
Comment (geoperez) [inline src/Grid/Paginator.js]: this is painful. Add  ({ from, to, count }) as function signature and then the ternary op.

File: src/Grid/RemoteDataSource.js [hunk 5]
```diff
@@ -81,8 +81,13 @@ class RemoteDataSource {
       
         return obj;
       });
-
-      this.dataStream.onNext({ Payload: rows, Aggregate: response.data.AggregationPayload });
+  
+      this.dataStream.onNext({ 
+        Payload: rows,
```
Comment (geoperez) [inline src/Grid/RemoteDataSource.js]: We should change this to camelcase the property name on the `dataStream `next  object

File: src/Grid/TablePaginationActions.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+class TablePaginationActions extends React.Component {
+  state = {
+    pages: [0, 1, 2, 3, 4],
+    activePage: 0
+  }
+
+  handleFirstPageButtonClick = event => {
+    this.props.onChangePage(event, 0);
+
+    this.setCurrentPage(0);
```
Comment (geoperez) [inline src/Grid/TablePaginationActions.js]: The `setCurrentPage` maybe should call `this.props.onChangePage` to reduce duplicated code.

File: src/Grid/GridHeader.js [hunk 7]
```diff
@@ -103,14 +99,65 @@ class GridHeader extends React.Component {
...
+    } 
+  }
+
+  localStorageHandler = count => {
+    const storage = JSON.parse(localStorage.getItem(`tubular.${this.props.gridName}`));
+    const dataSource = this.props.dataSource;
+
+    dataSource.columns = storage;
+
+    if(count < dataSource.columns.length) {
```
Comment (geoperez) [inline src/Grid/GridHeader.js]: Inverse and quick return.

File: example/src/main.js [hunk 8]
```diff
@@ -83,16 +83,37 @@ const columns = [
...
 export default class Main extends React.Component {
   render() {
     const dataSource = new RemoteDataSource('http://tubular.azurewebsites.net/api/orders/paged', columns);
-
+    
     return (
       <Grid dataSource={dataSource} 
-        bodyRenderer = {
+        gridName = 'Motorhead'
+        /* bodyRenderer = {
```
Comment (geoperez) [inline example/src/main.js]: Remove commented code.

**Pull request merged**